exporter/exporterhelper/internal/request.go

```diff
@@ -3,34 +3,29 @@
 
 package internal // import "go.opentelemetry.io/collector/exporter/exporterhelper/internal"
 
-import "context"
+import (
+	"context"
 
-// Request defines capabilities required for persistent storage of a request
-type Request interface {
-	// Context returns the context.Context of the requests.
-	Context() context.Context
+	"go.opentelemetry.io/collector/exporter/exporterhelper/request"
+)
 
-	// SetContext updates the context.Context of the requests.
-	SetContext(context.Context)
-
-	Export(ctx context.Context) error
-
-	// OnError returns a new Request may contain the items left to be sent if some items failed to process and can be retried.
-	// Otherwise, it should return the original Request.
-	OnError(error) Request
-
-	// Count returns the count of spans/metric points or log records.
-	Count() int
-
-	// OnProcessingFinished calls the optional callback function to handle cleanup after all processing is finished
-	OnProcessingFinished()
-
-	// SetOnProcessingFinished allows to set an optional callback function to do the cleanup (e.g. remove the item from persistent queue)
-	SetOnProcessingFinished(callback func())
+// QueueRequest defines a request coming through a queue.
+type QueueRequest struct {
+	request.Request
+	ctx                      context.Context
+	onProcessingFinishedFunc func()
 }
 
-// RequestUnmarshaler defines a function which takes a byte slice and unmarshals it into a relevant request
-type RequestUnmarshaler func([]byte) (Request, error)
+// OnProcessingFinished calls the optional callback function to handle cleanup after all processing is finished
+func (qr *QueueRequest) OnProcessingFinished() {
+	if qr.onProcessingFinishedFunc != nil {
+		qr.onProcessingFinishedFunc()
+	}
+}
 
-// RequestMarshaler defines a function which takes a request and marshals it into a byte slice
-type RequestMarshaler func(Request) ([]byte, error)
+func (qr *QueueRequest) Context() context.Context {
```

Comment:
Why do you need this if you can access the internal member?

Reply:
This is being used by the queue sender for sending the context to the next sender

Reply:
Why not passing it in the callback?

Reply:
I tried that. In that case, the QueueRequest is left with `onProcessingFinishedFunc` func only. So we can pass them as three fields, but the callback function gets pretty complicated.

```
Callback func(ctx context.Context, req request.Request, onProcessingFinishedFunc func())
```

Not sure if keeping the onProcessingFinishedFunc field only warrants having the struct like 

```
type QueueRequest struct {
	request.Request
	onProcessingFinishedFunc func()
}
```

Let me know WDYT

Reply:
Maybe keep `onProcessingFinishedFunc` as a prop on the Request and wrap the original request in case of the PersistenQueue?

Reply:
This is essentially the same as what's implemented in this PR, just with removed Context field, right?

```
type QueueRequest struct {
	request.Request
	onProcessingFinishedFunc func()
}
```

Reply:
Yes and now. Also change https://github.com/open-telemetry/opentelemetry-collector/pull/8867/files#diff-667c34fb5331f087f510bdf2b6f39e626e6a9e6d9003425b26236b660e3a5be2R17 to be `Callback func(context.Context, Request)`

Reply:
I realized that we need the context to be embedded for the channel interface we want to have in https://github.com/open-telemetry/opentelemetry-collector/pull/8828